lib/streamlit/elements/arrow_vega_lite.py

```diff
@@ -112,10 +113,10 @@ def vega_lite_chart(
         spec: dict[str, Any] | None = None,
         use_container_width: bool = False,
         theme: Literal["streamlit"] | None = "streamlit",
-        on_select: Union[str, Callable[..., None], None] = None,
+        on_select: Union[str, Callable[..., None], bool, None] = None,
         key: str | None = None,
         **kwargs: Any,
-    ) -> DeltaGenerator:
+    ) -> Union[DeltaGenerator, Dict[Any, Any]]:
```

Comment:
Same comment as above, we probably want to use an function overload here for a more correct typing.